shell/platform/darwin/ios/framework/Source/FlutterEngineTest.mm

```diff
@@ -30,6 +31,24 @@ - (void)testCreate {
   XCTAssertNotNil(engine);
 }
 
+- (void)testInfoPlist {
+  // Check the embedded Flutter.framework Info.plist, not the linked dylib.
+  NSURL* flutterFrameworkURL =
+      [NSBundle.mainBundle.privateFrameworksURL URLByAppendingPathComponent:@"Flutter.framework"];
```

Comment:
Instead of attempting to resolve the bundle by path, `[NSBundle bundleForClass:[FlutterEngine class]]` might be less brittle. For instance, I am not sure why this is in the private frameworks URL (presumably because its in the test bundle but not sure).

Reply:
I tried `-[NSBundle bundleForClass:]` first, unfortunately it doesn't work, the bundle is nil.  Before this PR, the test does not link against/embed/load the Flutter.framework with the Info.plist.  Instead, it embeds libios_test_flutter, which has the compiled Flutter and test code.
 https://github.com/flutter/engine/blob/0292339eb1a970c5a70c38abb7ce836824c54019/shell/platform/darwin/ios/BUILD.gn#L232
https://github.com/flutter/engine/blob/266435dafee3339283ecaed2b3bf1f34525c8748/testing/ios/IosUnitTests/IosUnitTests.xcodeproj/project.pbxproj#L39

```
$ nm -g out/ios_debug_sim_unopt/libios_test_flutter.dylib | grep FlutterEngine
0000000003987fd8 S _OBJC_CLASS_$_FlutterEngine
0000000003988078 S _OBJC_CLASS_$_FlutterEngineGroup
0000000003987880 S _OBJC_CLASS_$_FlutterEngineGroupTest
0000000003987ba0 S _OBJC_CLASS_$_FlutterEnginePartialMock
00000000039878d0 S _OBJC_CLASS_$_FlutterEngineTest
0000000003988028 S _OBJC_METACLASS_$_FlutterEngine
00000000039880a0 S _OBJC_METACLASS_$_FlutterEngineGroup
0000000003987858 S _OBJC_METACLASS_$_FlutterEngineGroupTest
0000000003987b78 S _OBJC_METACLASS_$_FlutterEnginePartialMock
00000000039878a8 S _OBJC_METACLASS_$_FlutterEngineTest
```

But I want to test the Info.plist in the final bundle, so I embedded it in the app in this PR so I could leverage the `NSBundle` API.  I'm open to suggestions for a better way to test this.